lib/initial_app.dart

```diff
@@ -6,39 +6,55 @@ import 'package:hatspace/features/sign_up/view_model/sign_up_bloc.dart';
 import 'package:hatspace/models/authentication/authentication_bloc.dart';
 import 'package:hatspace/strings/l10n.dart';
 import 'package:hatspace/theme/hs_theme.dart';
+import 'package:hatspace/view_models/app_config/bloc/app_config_bloc.dart';
+import 'package:hatspace/view_models/app_config/bloc/app_config_event.dart';
 
 class MyApp extends StatelessWidget {
   const MyApp({super.key});
 
   // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
-    return MultiBlocProvider(
-        providers: [
-          BlocProvider<AuthenticationBloc>(
-            create: (context) => AuthenticationBloc(),
-          ),
-          BlocProvider<SignUpBloc>(create: (context) {
-            return SignUpBloc()..add(const CheckFirstLaunchSignUp());
-          })
-          // TODO add your bloc creation here
-        ],
-        child: MaterialApp(
-          theme: lightThemeData,
-          home: BlocBuilder<SignUpBloc, SignUpState>(
-            builder: ((context, state) {
-              if (state is FirstLaunchScreen && state.isFirstLaunch == true) {
-                return const SignUpScreen();
-              } else {
-                return HomePageView();
-              }
-            }),
-          ),
-          debugShowCheckedModeBanner: false,
-          localizationsDelegates: const <LocalizationsDelegate<dynamic>>[
-            HatSpaceStrings.delegate
-          ],
-          supportedLocales: HatSpaceStrings.delegate.supportedLocales,
-        ));
+    return MultiBlocProvider(providers: [
+      BlocProvider<AppConfigBloc>(create: (context) {
+        return AppConfigBloc()..add(const OnInitialRemoteConfig());
+      }),
+      BlocProvider<AuthenticationBloc>(
+        create: (context) => AuthenticationBloc(),
+      ),
+      BlocProvider<SignUpBloc>(create: (context) {
+        return SignUpBloc()..add(const CheckFirstLaunchSignUp());
+      }),
+      // TODO add your bloc creation here
+    ], child: const MyAppBody());
+  }
+}
+
+class MyAppBody extends StatelessWidget {
+  const MyAppBody({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: lightThemeData,
+      home: BlocListener<SignUpBloc, SignUpState>(
```

Comment:
Update: 
Using Bloclistener to listen state of FirstLaunchApp to render Sign Up Screen.
-> otherwise, with initial state, HomeScreen will always be rendered first

Reply:
Also, This change will fix the bug that cause the application incorrect flow between sign up view and home view